Testing/VelaptorTests/Graphics/Renderers/FontRendererTests.cs

```diff
@@ -69,45 +70,53 @@ public class FontRendererTests : TestsBase
     /// </summary>
     public FontRendererTests()
     {
-        this.mockGL = new Mock<IGLInvoker>();
+        this.mockGL = Substitute.For<IGLInvoker>();
 
-        this.mockGLService = new Mock<IOpenGLService>();
-        this.mockGLService.Setup(m => m.ProgramLinkedSuccessfully(It.IsAny<uint>())).Returns(true);
-        this.mockGLService.Setup(m => m.ShaderCompiledSuccessfully(It.IsAny<uint>())).Returns(true);
-        this.mockGLService.Setup(m => m.GetViewPortSize()).Returns(new Size(800, 600));
+        this.mockGLService = Substitute.For<IOpenGLService>();
+        this.mockGLService.ProgramLinkedSuccessfully(Arg.Any<uint>()).Returns(true);
+        this.mockGLService.ShaderCompiledSuccessfully(Arg.Any<uint>()).Returns(true);
+        this.mockGLService.GetViewPortSize().Returns(new Size(800, 600));
 
-        this.mockShader = new Mock<IShaderProgram>();
-        this.mockShader.SetupGet(p => p.ShaderId).Returns(FontShaderId);
+        this.mockShader = Substitute.For<IShaderProgram>();
+        this.mockShader.ShaderId.Returns(FontShaderId);
 
-        this.mockGpuBuffer = new Mock<IGpuBuffer<FontGlyphBatchItem>>();
+        this.mockGpuBuffer = Substitute.For<IGpuBuffer<FontGlyphBatchItem>>();
 
-        this.mockBatchingManager = new Mock<IBatchingManager>();
+        this.mockBatchingManager = Substitute.For<IBatchingManager>();
 
-        var mockPushReactable = new Mock<IPushReactable>();
-        mockPushReactable.Setup(m => m.Subscribe(It.IsAny<IReceiveSubscription>()))
-            .Callback<IReceiveSubscription>(reactor => this.batchHasBegunReactor = reactor);
+        var mockPushReactable = Substitute.For<IPushReactable>();
+        mockPushReactable
+            .When(x => x.Subscribe(Arg.Any<IReceiveSubscription>()))
+            .Do(ci =>
+            {
+                var reactor = ci.Arg<IReceiveSubscription>();
+                this.batchHasBegunReactor = reactor;
+            });
 
-        var mockFontRenderBatchReactable = new Mock<IRenderBatchReactable<FontGlyphBatchItem>>();
+        var mockFontRenderBatchReactable = Substitute.For<RenderBatchReactable<FontGlyphBatchItem>>();
```

Comment:
Please change to what is shown below:

```suggestion
        var mockFontRenderBatchReactable = Substitute.For<IRenderBatchReactable<FontGlyphBatchItem>>();
```

**Reason:**
When it comes to classes, NSubstitute (and most of mocking frameworks) cannot mock a class that is set to 'sealed'.  The class `RenderBatchReactable` is a sealed class so this is the reason for all of the tests failing for this test class.

Changing this to `IRenderBatchReactable` will fix this issue.

Reply:
Done.